packages/cozy-doctypes/src/Document.js

```diff
@@ -152,12 +152,69 @@ class Document {
 
   static async createOrUpdate(attributes) {
     if (this.usesCozyClient()) {
-      throw new Error('This method is not implemented yet with CozyClient')
+      return this.createOrUpdateViaNewClient(attributes)
     }
 
     return this.createOrUpdateViaOldClient(attributes)
   }
 
+  static async createOrUpdateViaNewClient(attributes) {
+    const selector = fromPairs(
+      this.idAttributes.map(idAttribute => [
+        idAttribute,
+        get(attributes, sanitizeKey(idAttribute))
+      ])
+    )
+    let results = []
+    const compactedSelector = withoutUndefined(selector)
+    if (size(compactedSelector) === this.idAttributes.length) {
+      results = await this.queryAll(selector)
```

Comment:
I am worried by this method, it means that everytime we create a document, we need to query all the documents ?